tests/OffsetResultTest.php

```
@@ -165,4 +145,251 @@ public function testError()
+        ];
+
+        $offsetResult = new OffsetResult($this->getGenerator($sources));
+
+        // Memory check - fetch items one by one
+        $memoryBefore = memory_get_usage();
+        $count = 0;
+
+        while (($item = $offsetResult->fetch()) !== null) {
+            $count++;
+            // Simulate some processing
+            $processed = $item * 2;
```

PHPMD (2.15.0)

Avoid unused local variables such as '$processed'. (undefined)

(UnusedLocalVariable)